src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilterElement.java

```diff
@@ -74,13 +74,22 @@ private boolean isFileNameMatching(AuditEvent event) {
     private boolean isLineMatching(AuditEvent event) {
         boolean result = false;
         if (event.getLocalizedMessage() != null) {
-            for (List<Integer> aLineRangeList : lineRangeList) {
-                final int startLine = aLineRangeList.get(0) + 1;
-                final int endLine = aLineRangeList.get(1) + 1;
-                final int currentLine = event.getLine();
-                result = currentLine >= startLine && currentLine < endLine;
-                if (result) {
-                    break;
+            final String[] checkNames = event.getLocalizedMessage().getSourceName().split("\\.");
+            final String checkShortName = checkNames[checkNames.length - 1];
+            if ("RegexpOnFilenameCheck".equals(checkShortName)
+                    || "FileLengthCheck".equals(checkShortName)
+                    || "NewlineAtEndOfFileCheck".equals(checkShortName)) {
+                result = true;
+            }
+            else {
+                for (List<Integer> aLineRangeList : lineRangeList) {
```

Comment:
you can use stream api here

Smth like
```
return aLineRangeList.stream().any(lineRange -> {...})
```

Reply:
@timurt 
Is like this one, but I think it is less readable.
```java
private boolean isLineMatching(AuditEvent event) {
        boolean result = false;
        if (event.getLocalizedMessage() != null) {
            for (List<Integer> aLineRangeList : lineRangeList) {
                final int currentLine = event.getLine();
               result =
                       (aLineRangeList.stream().anyMatch(position -> currentLine >= position + 1 )
                           && aLineRangeList.stream().anyMatch(position -> currentLine < position + 1));
//                final int startLine = aLineRangeList.get(0) + 1;
//                final int endLine = aLineRangeList.get(1) + 1;
//                result = currentLine >= startLine && currentLine < endLine;
                if (result) {
                    break;
                }
            }
        }
        return result;
    }
```